server.py

```diff
@@ -32,8 +32,31 @@ def index(env):
 
 def login(env):
 
-    if env['REQUEST_METHOD'] == 'GET':
-        return 302, {'Location': META['rp_origin'] + '/'}, b''
+    if env['REQUEST_METHOD'] != 'POST':
+        return 400, {}, b''
+
+    body = env['wsgi.input'].read(int(env['CONTENT_LENGTH']))
+    email = parse.parse_qs(body)[b'email'][0].decode('ascii')
+
+    auth_url = '%(portier_origin)s/auth?%(qs)s' % {
```

Comment:
The way you're using formatting strings here seems kind of weird. The `%(name)s` style seems useful mainly if (a) you already have a dict containing the variable's names, or (b) if there are lots of variables being inserted. Neither of these is the case here, so it seems to me that just using `%s` would probably be more readable, and would prevent the silly repetition of `'portier_origin': META['portier_origin']`. `qs` can just be a local variable, after which you can do the formatting in place in the `return` statement, I think.